adapters/connectad/connectad.go

```diff
@@ -0,0 +1,213 @@
+package connectad
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type ConnectAdAdapter struct {
+	endpoint string
+}
+
+type connectadImpExt struct {
+	ConnectAd openrtb_ext.ExtImpConnectAd `json:"connectad"`
+}
+
+func (a *ConnectAdAdapter) SkipNoCookies() bool {
+	return false
+}
+
+func NewConnectAdBidder(endpoint string) *ConnectAdAdapter {
+	return &ConnectAdAdapter{
+		endpoint: endpoint,
+	}
+}
+
+func (a *ConnectAdAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	var errs []error
+
+	if len(request.Imp) == 0 {
```

Comment:
You can leave this check in here if you wish, other adapters do.. but it shouldn't be necessary. The core code will not call `MakeRequests` with no impressions.